bootstrap/handlers/auth_middleware.go

```diff
@@ -54,28 +60,32 @@ func VaultAuthenticationHandlerFunc(secretProvider interfaces.SecretProviderExt,
 				validToken, err := secretProvider.IsJWTValid(token)
 				if err != nil {
 					lc.Errorf("Error checking JWT validity: %v", err)
-					http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
-					return
+					// set Response.Committed to true in order to rewrite the status code
+					w.Committed = false
+					return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
```

Comment:
This one is still potentially leaking a lot of information.

Reply:
Thanks @bnevis-i for catching this place as well! Change updated.